pkg/csi/cinder/nodeserver.go

```diff
@@ -523,7 +523,16 @@ func (ns *nodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 	if len(volumeID) == 0 {
 		return nil, status.Error(codes.InvalidArgument, "Volume ID not provided")
 	}
-	volumePath := req.GetVolumePath()
+
+	// Get volume path
+	// With Kubernetes version>=v1.19.0, expand request carries volume_path and
+	// staging_target_path, what csi requires is staging_target_path.
+	volumePath := req.GetStagingTargetPath()
+	if volumePath == "" {
+		// If Kubernetes version < v1.19.0 the volume_path would be
+		// having the staging_target_path information
+		volumePath = req.GetVolumePath()
+	}
```

Comment:
@Fedosin could you explain your issue in more detail . 
NodeExpandVolume can be called after nodePublish , in that case,  in that case we cant use staging_target_path, volumepath stores the right value. please check here: https://github.com/gnufied/kubernetes/blob/master/pkg/volume/csi/csi_client.go#L101

It would be more helpful if you could log the issue with details  and the values you are receiving in volumePath and staging_target_path